### PR TITLE
 IMAGE: Move bitmap writing code out of OpenGLGraphicsManager

### PR DESCRIPTION
**Pull request by ccawley2011**
This hasn't been tested on big endian machines, although I suspect additional changes will be needed.

### CODE REVIEW

File: backends/graphics/opengl/opengl-graphics.cpp [hunk 1]
```diff
@@ -1212,41 +1214,13 @@ bool OpenGLGraphicsManager::saveScreenshot(const Common::String &filename) const
 	pixels.resize(lineSize * height);
 	GL_CALL(glReadPixels(0, 0, width, height, GL_RGB, GL_UNSIGNED_BYTE, &pixels.front()));
 
-#ifdef USE_PNG
 	const Graphics::PixelFormat format(3, 8, 8, 8, 0, 16, 8, 0, 0);
```
Comment (bgK) [inline backends/graphics/opengl/opengl-graphics.cpp]: Are you sure this is endian-safe? It seems to me `glReadPixels(.., GL_RGB, GL_UNSIGNED_BYTE, ..)` gives us pixels in RGB byte order. But `Graphics::PixelFormat(3, 8, 8, 8, 0, 16, 8, 0, 0);` is RGB in word order. I would expect something like:
```cpp
#ifdef SCUMM_LITTLE_ENDIAN
const Graphics::PixelFormat format(3, 8, 8, 8, 0, 16, 8, 0, 0);
#else
const Graphics::PixelFormat format(3, 8, 8, 8, 0, 0, 8, 16, 0);
#endif
```

Comment (ccawley2011) [reply backends/graphics/opengl/opengl-graphics.cpp]: Does that mean that saving PNG screenshots is also broken on big endian?

Comment (bgK) [reply backends/graphics/opengl/opengl-graphics.cpp]: I think PNG worked on BE by accident. See:
https://github.com/scummvm/scummvm/blob/22f12e5ffb91583a0955d0e8a637aff1089459d2/image/png.cpp#L266

`PNG_COLOR_TYPE_RGB` means the PNG encoder expects byte order RGB pixels. Which happens to be exactly what OpenGL provides. The PixelFormat is incorrect but the actual pixel data is correct.

File: image/png.cpp [hunk 2]
```diff
@@ -244,8 +244,13 @@ bool PNGDecoder::loadStream(Common::SeekableReadStream &stream) {
 
 bool writePNG(Common::WriteStream &out, const Graphics::Surface &input, const bool bottomUp) {
 #ifdef USE_PNG
-	const Graphics::PixelFormat requiredFormat_3byte(3, 8, 8, 8, 0, 16, 8, 0, 0);
+#ifdef SCUMM_LITTLE_ENDIAN
+	const Graphics::PixelFormat requiredFormat_3byte(3, 8, 8, 8, 0, 0, 8, 16, 0);
 	const Graphics::PixelFormat requiredFormat_4byte(4, 8, 8, 8, 8, 0, 8, 16, 24);
+#else
+	const Graphics::PixelFormat requiredFormat_3byte(3, 8, 8, 8, 0, 16, 8, 0, 0);
+	const Graphics::PixelFormat requiredFormat_4byte(4, 8, 8, 8, 8, 16, 8, 0, 24);
```
Comment (bgK) [inline image/png.cpp]: I believe this should be `const Graphics::PixelFormat requiredFormat_4byte(4, 8, 8, 8, 8, 24, 16, 8, 0)` as `PNG_COLOR_TYPE_RGB_ALPHA` requires pixels to be in RGBA byte order.

**Comment by bgK** [comment]
Thanks! I've verified this to work on a PlayStation 3 (PPC64 BE), merging.

**Pull request merged**